client/task-list/tasks/products/product-template-modal.js

```diff
@@ -104,36 +98,25 @@ export default function ProductTemplateModal( { onClose } ) {
 		>
 			<div className="woocommerce-product-template-modal__wrapper">
 				<div className="woocommerce-product-template-modal__list">
-					<List items={ templates }>
-						{ ( item, index ) => (
-							<div className="woocommerce-list__item-inner">
-								<input
-									id={ `product-templates-${
-										item.key || index
-									}` }
-									className="components-radio-control__input"
-									type="radio"
-									name="product-template-options"
-									value={ item.key }
-									onChange={ onSelectTemplateClick }
-									checked={ item.key === selectedTemplate }
-								/>
-								<label
-									className="woocommerce-list__item-text"
-									htmlFor={ `product-templates-${
-										item.key || index
-									}` }
-								>
-									<div className="woocommerce-list__item-label">
-										{ item.title }
-									</div>
-									<div className="woocommerce-list__item-subtitle">
-										{ item.subtitle }
-									</div>
-								</label>
-							</div>
-						) }
-					</List>
+					<RadioControl
+						selected={ selectedTemplate }
+						options={ templates.map( ( item ) => {
+							return {
+								label: (
+									<>
+										<div className="woocommerce-list__item-label">
```

Comment:
It feels "wrong" to use the `woocommerce-list__item-label` class here, as this isn't the intended use of the class. I think it would be better to either create a specific class for this, or even better, use the `Text` component from `@wordpress/components`.